clab/dependency_manager/dependency_manager.go

```diff
@@ -104,9 +104,21 @@ type defaultDependencyManager struct {
 
 // NewDependencyManager constructor.
 func NewDependencyManager() DependencyManager {
-	return &defaultDependencyManager{
+	dm := &defaultDependencyManager{
 		nodes: map[string]*dependencyNode{},
 	}
+
+	// init helper nodes
+	// TODO, we need to rework this stuff
+	for _, n := range []string{"macvlan", "macvtap"} {
+		dm.AddNode(n)
+		// artificially indicate that all the helper nodes have reached all the defined states
+		for _, s := range RegularNodeStates {
+			dm.SignalDone(n, s)
+		}
+	}
+
+	return dm
```

Comment:
@steiler why is this needed to support macvlan, but was not needed to support `host` kind?

If there is a difference, then maybe we can use `host` kind for macvlan/tap as well and not touch depmgr?

Reply:
you are right, there is always multiple ways towards rome.
For `host` and `mgmt-net` we prevented this dependency from getting inserted.
See https://github.com/srl-labs/containerlab/blob/main/clab/clab.go#L505
I'll extend that filter to also exclude the macvtap and macvlan (fake) nodes.

Reply:
either I removed this with a wrong force push, or you never added this condition?

Reply:
yepp you trashed 3 commits

Reply:
cherry-picked and pushed them now